src/circllhist.c

```diff
@@ -512,6 +512,84 @@ hist_approx_sum(const histogram_t *hist) {
   return sum;
 }
 
+double
+hist_approx_stddev(const histogram_t *hist) {
+  int i;
+  double total_count = 0.0;
+  double s1 = 0.0;
+  double s2 = 0.0;
+  ASSERT_GOOD_HIST(hist);
+  if(hist->used == 0) return 0.0;
+  for(i=0; i<hist->used; i++) {
+    if(hist_bucket_isnan(hist->bvs[i].bucket)) continue;
+    double midpoint = hist_bucket_midpoint(hist->bvs[i].bucket);
+    double count = hist->bvs[i].count;
+    total_count += count;
+    s1 += midpoint * count;
+    s2 += pow(midpoint, 2.0) * count;
+  }
+  assert(total_count > 0.0);
+  return sqrt(s2 / total_count - pow(s1 / total_count, 2.0));
+}
+
+double
+hist_approx_moment(const histogram_t *hist, double k) {
+  int i;
+  double total_count = 0.0;
+  double sk = 0.0;
+  ASSERT_GOOD_HIST(hist);
+  for(i=0; i<hist->used; i++) {
+    if(hist_bucket_isnan(hist->bvs[i].bucket)) continue;
+    double midpoint = hist_bucket_midpoint(hist->bvs[i].bucket);
+    double count = hist->bvs[i].count;
+    total_count += count;
+    sk += pow(midpoint, k) * count;
+  }
```

Comment:
What happens here if we have on item in the histogram that is NaN?

total_count  == 0?

Reply:
- total_count would be incremented as usual, and count the number samples in the NaN bucket.
- But `sk` would become NaN, since midpoint would be NaN.

This is probably not what we want here. I vote for ignoring samples in the NaN bucket here as well, so:
- Don't update total_count
- Don't update sk
in case we have a NaN bucket (i.e. `hist_bucket_isnan(hist->bvs[i])`)

Reply:
Sorry. I missed line 542: `if(hist_bucket_isnan(hist->bvs[i].bucket)) continue;`

So yes. total_count would be 0, and we should return NaN, since we can't divide by 0.